src/System.Text.Primitives/System/Text/InvariantParser_bool.cs

```diff
@@ -0,0 +1,213 @@
+            if (secondChar != 'a' && secondChar != 'A')
+                return false;
+
+            byte thirdChar = utf8Text[index + 2];
+            if (thirdChar != 'l' && thirdChar != 'L')
+                return false;
+
+            byte fourthChar = utf8Text[index + 3];
+            if (fourthChar != 's' && fourthChar != 'S')
+                return false;
+
+            byte fifthChar = utf8Text[index + 4];
+            if (fifthChar != 'e' && fifthChar != 'E')
+                return false;
+
+            return true;
+        }
+        private unsafe static bool isFalse(byte* utf8Text, int index, int length)
+        {
+            if (length < 5)
+                return false;
+
+            byte firstChar = utf8Text[index];
+            if (firstChar != 'f' && firstChar != 'F')
+                return false;
+
+            byte secondChar = utf8Text[index + 1];
+            if (secondChar != 'a' && secondChar != 'A')
+                return false;
+
+            byte thirdChar = utf8Text[index + 2];
+            if (thirdChar != 'l' && thirdChar != 'L')
+                return false;
+
+            byte fourthChar = utf8Text[index + 3];
+            if (fourthChar != 's' && fourthChar != 'S')
+                return false;
+
+            byte fifthChar = utf8Text[index + 4];
+            if (fifthChar != 'e' && fifthChar != 'E')
+                return false;
+
+            return true;
+        }
+        #endregion
+
+        public static bool TryParse(byte[] utf8Text, int index, out bool value, out int bytesConsumed)
+        {
+            bytesConsumed = 0;
+            value = default(bool);
+
+            if (utf8Text.Length < 1 || index < 0 || index >= utf8Text.Length)
+            {
+                return false;
+            }
+
+            byte firstByte = utf8Text[index];
+
+            if (firstByte == '1')
+            {
+                bytesConsumed = 1;
+                value = true;
+                return true;
+            }
+            else if (firstByte == '0')
+            {
+                bytesConsumed = 1;
+                value = false;
+                return true;
+            }
+            else if (isTrue(utf8Text, index))
+            {
+                bytesConsumed = 4;
+                value = true;
+                return true;
+            }
+            else if (isFalse(utf8Text, index))
+            {
+                bytesConsumed = 5;
+                value = false;
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        public static unsafe bool TryParse(byte* utf8Text, int index, int length, out bool value, out int bytesConsumed)
+        {
+            bytesConsumed = 0;
+            value = default(bool);
+
+            if (length < 1 || index < 0)
+            {
+                return false;
+            }
+
+            byte firstByte = utf8Text[index];
```

Comment:
oh, thank you. I had done some refactoring and apparently ended up with duplicate code I didn't notice.